src/ert/storage/migration/to4.py

```diff
@@ -20,8 +20,7 @@ def migrate(path: Path) -> None:
         if observations:
             output_path = experiment / "observations"
             output_path.mkdir(parents=True, exist_ok=True)
-            for name, dataset in observations.items():
-                dataset.to_netcdf(output_path / f"{name}", engine="scipy")
+            ert_config.write_observations_to_folder(dest=output_path)
```

Comment:
The only problem with this is in cases where we need to migrate the existing observation files, if they are `current` and a later migration tries to migrate then, that will cause problems 🤔

Reply:
Yep, though only `to7.py`  at the moment touches observation files, where we have this check:
```
    for experiment in path.glob("experiments/*"):
        _obs_keys = os.listdir(os.path.join(experiment, "observations"))

        if len(set(_obs_keys) - {"summary", "gen_data"}) == 0:
            # Observations are already migrated, likely from .to4 migrations
            continue
```

So it will skip the migration if already migrated, and migrate if it needs migration. Might cause more trouble in the future if we do more changes though